src/mpid/ch3/src/mpid_vc.c

```diff
@@ -424,6 +424,225 @@ int MPID_GPID_ToLpidArray( int size, MPIR_Gpid in_gpid[], int lpid[] )
     return mpi_errno;
 }
 
+#undef FUNCNAME
+#define FUNCNAME MPIDI_LPID_GetAllInComm
+#undef FCNAME
+#define FCNAME MPL_QUOTE(FUNCNAME)
+static inline int MPIDI_LPID_GetAllInComm(MPIR_Comm *comm_ptr, int local_size,
+                                          int local_lpids[])
+{
+    int i;
+    MPIR_Assert( comm_ptr->local_size == local_size );
+    for (i=0; i<comm_ptr->local_size; i++) {
+	(void)MPID_Comm_get_lpid( comm_ptr, i, &local_lpids[i], FALSE );
+    }
+    return 0;
```

Comment:
I think actually checking the return of `MPID_Comm_get_lpid` is better. Something like `ret |= MPID_Comm_get_lpid( ... );` and `return ret;` would be sufficient. Note that `MPID_Comm_get_lpid()` always do `return MPI_SUCCESS`, so the compiler should eventually optimize it. I think `void` function is not the best choice because we will have to change the function definition again if we later decide to add error code here. Unused return value can be optimized by compiler.